### PR TITLE
Fix for Issue 190

### PR DESCRIPTION
**Pull request by christophernorth**
Fix Issue with Metropolis Hastings algorithm not generating samples correctly (we always returned the same value) if Downsampling not used when streaming.

Also includes a small fix to the Metropolis Hastings tests to make it more reliable (it was randomly failing for me).

### CODE REVIEW

**Comment by christophernorth** [comment]
Fixes #190

File: keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/MetropolisHastings.java [hunk 1]
```diff
@@ -142,6 +142,7 @@ public void sample(Map<Long, List<?>> samplesByVertex) {
 
         @Override
         public NetworkState sample() {
+            step();
```
Comment (icantsayrural) [inline keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/MetropolisHastings.java]: The doc in `SamplingAlgorithm#sample` is unclear to me.
```
    /**
     * @return a network state that represents the value of vertices at the
     * end of the algorithm step
     */
```
Whether or not it's being used to sample at the end of the algorithm step is outside of its scope, even more so if we add this in. Maybe it'll be better to change `NetworkSamplesGenerator#stream` to take one more step?

File: keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/SamplingAlgorithm.java [hunk 2]
```diff
@@ -8,20 +8,23 @@
...
 
     /**
-     * Same effect as a sample but the result isn't saved or returned.
+     * Move forward the state of the Sampling Algorithm by a single step.
      */
     void step();
 
     /**
-     * Takes a sample with the algorithm and saves it in the supplied map
+     * Takes a sample with the algorithm and saves it in the supplied map.  Repeated calls to this function will return
```
Comment (icantsayrural) [inline keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/SamplingAlgorithm.java]: > Repeated calls to this function will return the same values without an intermediary call to `step()`

I don't follow. This method doesn't explicitly return anything? And repeated calls will update the sample list for each vertex.

Comment (christophernorth) [reply keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/SamplingAlgorithm.java]: Good catch - I was documenting the wrong fix.  Made changes.

**Pull request merged**